test_crates/struct_with_no_pub_fields_changed/new/src/lib.rs

```diff
@@ -0,0 +1,104 @@
+pub enum PubStructChangedToEnum {
+    Foo(usize),
+}
+
+pub union PubStructChangedToUnion {
+    foo: usize
+}
+
+pub enum PubStructWithNonPubDocFieldChangedToEnum {
+    Foo(usize),
+}
+
+
+pub union PubStructWithNonPubDocFieldChangedToUnion {
+    /// Despite this field being pub, hiding it makes this not be `public_api_eligible` anymore
+    /// This struct should trigger `struct_with_no_pub_fields_changed` instead of `struct_with_pub_fields_changed`
```

Comment:
Lint name correction:
```suggestion
    /// This struct should trigger `struct_with_no_pub_fields_changed` instead of `struct_with_pub_fields_changed_type`
```

Please also add a ruler to your editor at 100 chars and try to not exceed it unless there's good reason. Hard enforcement of line lengths is annoying because e.g. some URLs are just long and there's no much we can do about it. But this line is 110+ chars wide and there's no reason why it has to be that long all on one line.